source/android/adaptivecards/src/main/java/io/adaptivecards/renderer/layout/HorizontalFlowLayout.java

```diff
@@ -232,4 +245,44 @@ int measureRequiredHeight(final int width,
 
         return ypos;
     }
+
+    /**
+     * Measures the width required by this view
+     * by summing the width and padding of all of it's children.
+     * <p>
+     * (Package-private visibility for uint tests access.)
+     *
+     * @param paddingLeft left padding for this view.
+     * @param paddingRight right padding for this view.
+     * @return the height required by this view.
+     */
+    int measureRequiredWidth(final int paddingLeft,
```

Comment:
does this ensure that images are set?
if the images are not yet loaded, will this method still work?
what is ensuring that images size are available?

Reply:
I debugged in the visualizer, and the images are rendered before calling onMeasure(). onMeasure is called when the card is displayed (switching to the Visual tab), and images are loaded as soon as the JSON is rendered.

This is also very similar logic to the `measureRequiredHeight` method, so I don't think it is an issue.

Reply:
or it could be that it works with the fast connection; and may fails with slower connection. but this should be fine since  that logic has been there for while.